network/network.go

```diff
@@ -180,6 +195,14 @@ func (n *Network) Start() error {
 		return err
 	}
 
+	// Sanity check for configured node DID: can we resolve it?
+	if n.configuredNodeDID != nil {
```

Comment:
You should also check for the existence of the private key!

Reply:
The question is... which one? We haven't spec'd that yet.

I think we should introduce a DID `keyAgreement` key here (and in the spec), because it will be used to decrypt the `to` header. We should probably check here whether the first `keyAgreement` key of the DID document is present?

Reply:
@woutslakhorst I added these checks